code/game/objects/structures/_structure_materials.dm

```diff
@@ -56,14 +56,22 @@
 
 /obj/structure/proc/create_dismantled_products(var/turf/T)
 	SHOULD_CALL_PARENT(TRUE)
-	if(parts_type)
-		new parts_type(T, (material && material.type), (reinf_material && reinf_material.type))
+	if(parts_type && !ispath(parts_type, /obj/item/stack))
+		for(var/i = 1 to max(parts_amount, 1))
+			new parts_type(T, (material && material.type), (reinf_material && reinf_material.type))
 	else
 		for(var/mat in matter)
 			var/decl/material/M = GET_DECL(mat)
-			var/placing = FLOOR((matter[mat] / SHEET_MATERIAL_AMOUNT) * 0.75)
+			var/placing
+			if(isnull(parts_amount))
+				placing = (matter[mat] / SHEET_MATERIAL_AMOUNT) * 0.75
+				if(parts_type)
+					placing *= atom_info_repository.get_matter_multiplier_for(parts_type, mat, placing)
+				placing = FLOOR(placing)
+			else
+				placing = parts_amount
 			if(placing > 0)
-				M.place_dismantled_product(T, amount = placing)
+				M.place_dismantled_product(T, FALSE, placing, parts_type)
```

Comment:
I feel it would make more sense to feed an absolute amount of matter into this proc, and then have it adjust the stack size produced so that matter is conserved. Alternatively, to wrap this proc in a variant that does what I describe (and what the logic above accomplishes, though perhaps without the caching). Is that radically incompatible with existing systems?

Reply:
That would also be feasible, but would largely just be shifting some code around. Structures are the only things that call this proc currently. Can do it if you like, not fussed.

Reply:
My worry is that having to reproduce the code above each time I want to rods is undesirable.

Reply:
Why would you need to?

Reply:
Having looked at the code I don't really want to move the stack logic onto the material. Currently `parts_amount` will bypass the matter value checking and caching and I will need to multiple that or do some other math to munge it into an appropriate matter amount.